app/lib/katello/resources/cdn.rb

```diff
@@ -66,7 +66,7 @@ def http_downloader
           # Run the following command in rails console to figure out other
           # valid constants in other ruby versions
           # "OpenSSL::SSL::SSLContext::METHODS"
-          net.ssl_version = SETTINGS[:katello][:cdn_ssl_version] if SETTINGS[:katello].key?(:cdn_ssl_version)
+          net.ssl_version = Setting[:cdn_ssl_version] if Setting[:cdn_ssl_version]
```

Comment:
Since we can't validate the value of setting when we set or modify it, we might want to perform some validation in this file where we use that value since something bogus is probably going to raise a not very helpful error.

One solution might be to add a constant in this file:

SUPPORTED_SSL_VERSIONS = ['SSLv23', 'TLSv1']

Then in the constructor we can check:

```
if Setting[:cdn_ssl_version] && !SUPPORTED_SSL_VERSIONS.include?(Setting[:cdn_ssl_version))
   fail("Invalid SSL version specified. Check the 'CDN SSL Version' setting")
end
```

Let me know if you like the idea. This should give a useful error when enabling a repository when the SSL version is misconfigured.

Reply:
There's a list of valid options, shouldn't that be sufficient validation?

Reply:
@ekohl i'm not sure how those options come in to play. in the UI I can still enter any value that I want and I don't see any reference to the valid options, even on the CLI

```
[vagrant@centos7-hammer-devel ~]$ hammer settings set --name cdn_ssl_version --value "whatever"
Setting [cdn_ssl_version] updated to [whatever].
```
```
vagrant@centos7-hammer-devel ~]$ hammer settings list | grep cdn_ssl_version
cdn_ssl_version                                        | SSL version used to communicate with the CDN                | whatever                                                                         | SSL version used to communicate with the CDN      
```

Unless I'm missing something I think the validation is useful because it appears there's none and the user does not even know the valid options

Reply:
Tested and found that setting some random value for SSL version doesn't seem to hurt anything when making the connection. I guess it's ignored and the version is renegotiated, so maybe the validation really is unnecessary. I feel less strongly about the validation now since it's not a total failure when the value is invalid.

Reply:
Hmm wait, now I am seeing a stack trace:

```
16:37:00 rails.1   | 2020-04-02T16:37:00 [W|not|23dac3ad] Failed to handle notifications - this is most likely a bug: unknown SSL method `whatever'                                     
16:37:00 rails.1   | 2020-04-02T16:37:00 [D|not|23dac3ad] /opt/rh/rh-ruby25/root/usr/share/ruby/openssl/ssl.rb:209:in `ssl_version='                                                          16:37:00 rails.1   |  | /opt/rh/rh-ruby25/root/usr/share/ruby/openssl/ssl.rb:140:in `block in set_params'                          
16:37:00 rails.1   |  | /opt/rh/rh-ruby25/root/usr/share/ruby/openssl/ssl.rb:140:in `each'                                                                                                    
16:37:00 rails.1   |  | /opt/rh/rh-ruby25/root/usr/share/ruby/openssl/ssl.rb:140:in `set_params'                                                                                              16:37:00 rails.1   |  | /opt/rh/rh-ruby25/root/usr/share/ruby/net/http.rb:971:in `connect'                                                                                                    
16:37:00 rails.1   |  | /opt/rh/rh-ruby25/root/usr/share/ruby/net/http.rb:920:in `do_start'                                                                                                   
16:37:00 rails.1   |  | /opt/rh/rh-ruby25/root/usr/share/ruby/net/http.rb:909:in `start'                                                                                                      
16:37:00 rails.1   |  | /home/vagrant/katello/app/lib/katello/resources/cdn.rb:94:in `get'
```

Reply:
I will put the validation in that you talked about then :)